### ISSUE TITLE
There is a bug when chaining path plot animation like so: path.animate().plot(p1).animate().plot(p2)

### ISSUE THREAD

**Issue by RmiTtro** [svgdotjs/svg.js#584]
The bug is that the second call to `plot` interfere with the first call to `plot`. This bug is caused by a combination of two factors. First, the patharray returned by [SVG.Path](https://github.com/svgdotjs/svg.js/blob/516f575cf4ea4921fab4d66fbce3b8dad25e7e66/src/path.js#L13-L19) is cached. Second, the method [plot](https://github.com/svgdotjs/svg.js/blob/516f575cf4ea4921fab4d66fbce3b8dad25e7e66/src/fx.js#L854) of the FX module call the method `morph` on the patharray returned by `SVG.Path` while ideally this should be done in the method `initAnimations`. 

Here is an in-deep explanation of what is happening:

```javascript
// path is an instance of SVG.Path
// p1 and p2 are SVG.PathArray
path.animate().plot(p1).animate().plot(p2)
```
In the first call to `plot`, `this.target().array()` is called which return an `SVG.PathArray`. This patharray is cached in `this.target()._array` where `this.target()` is an instance of `SVG.Path`. After that, `.morph(p1)` is called on that the patharray.

In the second call to `plot`, `this.target().array()` is called which return **the value that was cached previously**, it return the value in `this.target()._array`. After that, `.morph(p2)` is called on that patharray which as for effect of changing the destination of the first plot since it is use the same patharray.

That pretty much sums it up (I just hope my explanation wasn't too bad). As for solving this issue, I think that solving #547 should solve this issue. I will have a look at it!

**Comment by Fuzzyma** [comment]
I am a bit curious about this issue.
When `animate().plot()` is called the first time, `initAnimations()` gets the current `PathArray` and uses that for morphing. However: while animating this path, the cached PathArray is changed because `plot()` is called multiple times while animating the path. So I cant understand how the next call to `animate().plot()` (and therefore the call to `initAnimations()` again) can get the same PathArray than before.
Whenever you call plot with a value, the cached value is updated:
```js
return this.attr('d', (this._array = new SVG.PathArray(p)))
```

so this first cached array should not be equal to the second one.
Where is my mistake?

//EDIT: Oh I know see it. The problem really lays in obtaining the path array. Funny enough there is a comment in the code: 
```js
          // TODO: this is not a clean clone of the array. We may have some unchecked references
          s.animations[i].value = (i == 'plot' ? this.target().array().value : this.target()[i]())
```

Thats why I wanted to introduce a `clone()` method to every array object of svg because cloning is needed sometimes and doing it in place is noisy.

**Comment by RmiTtro** [comment]
I am not 100% sure but I think `initAnimations()` is only called after all the chain have been traversed. If you look the method [add()](https://github.com/svgdotjs/svg.js/blob/516f575cf4ea4921fab4d66fbce3b8dad25e7e66/src/fx.js#L491) you will see that it call `start()` (`start()` call `initAnimations()` eventually) like this: 

```javascript
setTimeout(function(){this.start()}.bind(this), 0)
```

Since `start()` is called using a `setTimeout()` and JavaScript is _mostly_ single-threaded I believe that `start()` (and by extension `initAnimations()`) do not get called right away. That is why when the second `plot` is reached he pick the same reference as the first one. Anyway, that is how I understood it, I could be wrong.

As for adding a `clone()` method to the arrays, I think that would be good idea. I must admit that is the first thing I looked for while searching a solution for this issue.

**Comment by Fuzzyma** [comment]
Fixed with #589
